gossip/emitter/throttling/dominant_set_test.go

```diff
@@ -0,0 +1,200 @@
+// Copyright 2025 Sonic Operations Ltd
+// This file is part of the Sonic Client
+//
+// Sonic is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Sonic is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with Sonic. If not, see <http://www.gnu.org/licenses/>.
+
+package throttling
+
+import (
+	"math/rand"
+	"slices"
+	"testing"
+
+	"github.com/Fantom-foundation/lachesis-base/inter/idx"
+	"github.com/Fantom-foundation/lachesis-base/inter/pos"
+	"github.com/stretchr/testify/require"
+)
+
+func TestComputeDominantSet_CanIdentifyWhenStakeDistributionIsDominated(t *testing.T) {
+
+	tests := map[string]struct {
+		stakes            []int64
+		expectedDominated bool
+		expectedSet       []idx.ValidatorID
+	}{
+		"no validators": {
+			stakes:            nil,
+			expectedDominated: false,
+		},
+		"single validator below threshold": {
+			stakes:            []int64{50},
+			expectedDominated: false,
+		},
+		"single validator just below threshold": {
+			stakes:            []int64{99},
+			expectedDominated: false,
+		},
+		"multiple validators below threshold": {
+			stakes:            append(slices.Repeat([]int64{10}, 9), 9),
+			expectedDominated: false,
+		},
+		"multiple validators summing threshold": {
+			stakes:            append(slices.Repeat([]int64{10}, 9), []int64{9, 1}...),
+			expectedDominated: true,
+			// all 11 elements are included
+			expectedSet: []idx.ValidatorID{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11},
+		},
+		"multiple validators just avobe threshold": {
+			stakes:            append(slices.Repeat([]int64{10}, 9), []int64{9, 1, 1}...),
+			expectedDominated: true,
+			//  11 elements out of 12 are included
+			expectedSet: []idx.ValidatorID{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11},
+		},
+		"single validator at threshold": {
+			stakes:            []int64{100},
+			expectedDominated: true,
+			expectedSet:       []idx.ValidatorID{1},
+		},
+		"single validator just above threshold": {
+			stakes:            []int64{101},
+			expectedDominated: true,
+			expectedSet:       []idx.ValidatorID{1},
+		},
+		"two validators at threshold": {
+			stakes:            []int64{50, 50},
+			expectedDominated: true,
+			expectedSet:       []idx.ValidatorID{1, 2},
+		},
+		"two validators at threshold out of three": {
```

Comment:
nit:
```suggestion
		"two validators at threshold above three": {
```
not sure which one reads better. wdyt?